From fb98a845d155fdfbd45c22a6b062c3cfbe692a0a Mon Sep 17 00:00:00 2001
From: Sebastian Kisela <skisela@redhat.com>
Date: Wed, 25 Jul 2018 09:18:33 +0200
Subject: [PATCH] major and minor functions moved to sysmacros.h

According to https://sourceware.org/git/?p=glibc.git;a=blob_plain;f=NEWS;hb=HEAD
(glibc changelog) the below happened, therefore explicitly include
<sys/sysmacros.h>

"
...
The macros 'major', 'minor', and 'makedev' are now only available from
the header <sys/sysmacros.h>; not from <sys/types.h> or various other
headers that happen to include <sys/types.h>.  These macros are rarely
used, not part of POSIX nor XSI, and their names frequently collide with
user code; see https://sourceware.org/bugzilla/show_bug.cgi?id=19239 for
further explanation.

<sys/sysmacros.h> is a GNU extension.  Portable programs that require
these macros should first include <sys/types.h>, and then include
<sys/sysmacros.h> if __GNU_LIBRARY__ is defined.
...
"

Signed-off-by: Sebastian Kisela <skisela@redhat.com>
---
 src/lockdev.c        | 1 +
 src/lockdev.c.access | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/lockdev.c b/src/lockdev.c
index 6e69894..ec86f65 100644
--- a/src/lockdev.c
+++ b/src/lockdev.c
@@ -121,6 +121,7 @@
 #include <sys/stat.h>
 #include <sys/file.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/wait.h>
 #include "lockdev.h"
 #include "ttylock.h"
diff --git a/src/lockdev.c.access b/src/lockdev.c.access
index 9a0fca5..3581938 100644
--- a/src/lockdev.c.access
+++ b/src/lockdev.c.access
@@ -117,6 +117,7 @@
 #include <sys/stat.h>
 #include <sys/file.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/wait.h>
 #include "lockdev.h"
 #include "ttylock.h"
-- 
2.14.4

